src/SDKs/Azure.ApplicationModel.Configuration/data-plane/Azure.ApplicationModel.Configuration/Options.cs

```diff
@@ -24,26 +25,49 @@ public enum SettingFields : uint
         All = uint.MaxValue
     }
 
-    public class RequestOptions
+    public class ConfigurationSelector
     {
+        public static readonly string Any = "*";
         /// <summary>
-        /// Specific label of the key.
+        /// Keys that will be used to filter.
         /// </summary>
-        public string Label { get; set; } = null;
-
-        public ETagFilter ETag { get; set; }
-
+        /// <remarks>See the documentation for this SDK for details on the format of filter expressions</remarks>
+        public List<string> Keys { get; set; } = new List<string> { Any };
         /// <summary>
-        /// If set, then key values will be retrieved exactly as they existed at the provided time.
+        /// Labels that will be used to filter.
         /// </summary>
-        public DateTimeOffset? Revision { get; set; }
-
+        /// <remarks>See the documentation for this SDK for details on the format of filter expressions</remarks>
+        public List<string> Labels { get; set; } = new List<string> { Any };
         /// <summary>
         /// IKeyValue fields that will be retrieved.
         /// </summary>
         public SettingFields Fields { get; set; } = SettingFields.All;
-        
-        public static implicit operator RequestOptions(string label) => new RequestOptions() { Label = label };
+        /// <summary>
+        /// If set, then key values will be retrieved exactly as they existed at the provided time.
+        /// </summary>
+        public DateTimeOffset? AcceptDateTime { get; set; }
+
+        public ConfigurationSelector() { }
+
+        public ConfigurationSelector(string key, string label = default)
+        {
+            Keys = new List<string> { key };
+            if(label != default) Labels = new List<string> { label };
```

Comment:
Does this set the capacity of the list to 1?

Reply:
Yes. I imagine if people want to pass more than one value, this wouldn't be the way to do it.
This is like the fast convenient way...

Reply:
Labels list is already created at this point, no need to recreate it, you can just call `Labels.Add(label)`.

Reply:
We should not eagerly create labels list, unless we think almost all settings will have labels. Having said that, I am ok with checking in as-is (after @pakrym's feedback fixed) and optimize it later.

Reply:
>Labels list is already created at this point,

By default it will include the wildcard Any. If I do `Labels.Add(label)` the result will be `{"*", "<label>"}` .

Reply:
`Labels[0] = label;` ?

Reply:
Or add the "*" in default ctor instead of inline.